tests/client/OperationDecorator/log_warning_for_deprecated_op_test.py

```diff
@@ -0,0 +1,39 @@
+from mock import patch
+import pytest
+
+from bravado_core.operation import Operation
+from bravado_core.spec import Spec
+
+from bravado.client import OperationDecorator
+
+
+@pytest.fixture
+def getPetById_spec(petstore_dict):
+    op = petstore_dict['paths']['/pet/{petId}']['get'].copy()
+    op.update({'deprecated': True,
```

Comment:
Additional test cases I can think of:
- test_deprecated_false()
- test_deprecated_not_present()